SWTError: No more handles

After running Peak Detector/Review UI in a batch job on Windows 10. The PC was also struggeling performance wise.

```
java.util.concurrent.ExecutionException: org.eclipse.swt.SWTError: No more handles
                at java.util.concurrent.FutureTask.report(FutureTask.java:122)
                at java.util.concurrent.FutureTask.get(FutureTask.java:192)
                at org.eclipse.chemclipse.support.ui.workbench.DisplayUtils.executeInUserInterfaceThread(DisplayUtils.java:103)
                at org.eclipse.chemclipse.support.ui.workbench.DisplayUtils.executeInUserInterfaceThread(DisplayUtils.java:82)
                at net.openchrom.xxd.process.supplier.templates.ui.core.PeakDetectorMSD.detect(PeakDetectorMSD.java:53)
                at org.eclipse.chemclipse.chromatogram.msd.peak.detector.core.PeakDetectorMSD.detect(PeakDetectorMSD.java:80)
                at org.eclipse.chemclipse.chromatogram.msd.peak.detector.core.PeakDetectorMSDProcessTypeSupplier$PeakDetectorProcessorSupplier.apply(PeakDetectorMSDProcessTypeSupplier.java:74)
                at org.eclipse.chemclipse.chromatogram.msd.peak.detector.core.PeakDetectorMSDProcessTypeSupplier$PeakDetectorProcessorSupplier.apply(PeakDetectorMSDProcessTypeSupplier.java:1)
                at org.eclipse.chemclipse.model.supplier.ChromatogramSelectionProcessorSupplier.apply(ChromatogramSelectionProcessorSupplier.java:39)
                at org.eclipse.chemclipse.model.supplier.IChromatogramSelectionProcessSupplier$1.execute(IChromatogramSelectionProcessSupplier.java:54)
                at org.eclipse.chemclipse.processing.supplier.IProcessSupplier.applyProcessor(IProcessSupplier.java:144)
                at org.eclipse.chemclipse.processing.methods.ProcessEntryContainer.applyProcessEntries(ProcessEntryContainer.java:190)
                at org.eclipse.chemclipse.processing.methods.ProcessEntryContainer.applyProcessEntries(ProcessEntryContainer.java:130)
                at org.eclipse.chemclipse.chromatogram.msd.process.supplier.batchprocess.core.BatchProcess.execute(BatchProcess.java:69)
                at org.eclipse.chemclipse.chromatogram.msd.process.supplier.batchprocess.ui.editors.BatchJobEditor.run(BatchJobEditor.java:196)
                at org.eclipse.jface.operation.ModalContext$ModalContextThread.run(ModalContext.java:119)
Caused by: org.eclipse.swt.SWTError: No more handles
                at org.eclipse.swt.SWT.error(SWT.java:4578)
                at org.eclipse.swt.SWT.error(SWT.java:4467)
                at org.eclipse.swt.SWT.error(SWT.java:4438)
                at org.eclipse.swt.widgets.Display.error(Display.java:1182)
                at org.eclipse.swt.widgets.Display.internal_new_GC(Display.java:2555)
                at org.eclipse.swt.graphics.Device.getDepth(Device.java:453)
                at org.eclipse.swt.widgets.Display.getIconDepth(Display.java:1829)
                at org.eclipse.swt.widgets.Decorations.setImages(Decorations.java:910)
                at org.eclipse.swt.widgets.Decorations.setImages(Decorations.java:998)
                at org.eclipse.jface.window.Window.configureShell(Window.java:368)
                at org.eclipse.jface.wizard.WizardDialog.configureShell(WizardDialog.java:466)
                at org.eclipse.jface.window.Window.createShell(Window.java:493)
                at org.eclipse.jface.window.Window.create(Window.java:425)
                at org.eclipse.jface.dialogs.Dialog.create(Dialog.java:1096)
                at org.eclipse.jface.window.Window.open(Window.java:783)
                at net.openchrom.xxd.process.supplier.templates.ui.wizards.PeakDetectorSupport.addPeaks(PeakDetectorSupport.java:53)
                at net.openchrom.xxd.process.supplier.templates.ui.core.PeakDetectorMSD$1.run(PeakDetectorMSD.java:60)
                at org.eclipse.chemclipse.support.ui.workbench.DisplayUtils$2.call(DisplayUtils.java:87)
                at org.eclipse.chemclipse.support.ui.workbench.DisplayUtils$2.call(DisplayUtils.java:1)
                at java.util.concurrent.FutureTask.run(FutureTask.java:266)
                at org.eclipse.swt.widgets.RunnableLock.run(RunnableLock.java:37)
                at org.eclipse.swt.widgets.Synchronizer.runAsyncMessages(Synchronizer.java:182)
                at org.eclipse.swt.widgets.Display.runAsyncMessages(Display.java:3906)
                at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:3537)
                at org.eclipse.jface.operation.ModalContext$ModalContextThread.block(ModalContext.java:165)
                at org.eclipse.jface.operation.ModalContext.run(ModalContext.java:369)
                at org.eclipse.jface.dialogs.ProgressMonitorDialog.run(ProgressMonitorDialog.java:471)
                at org.eclipse.chemclipse.ux.extension.xxd.ui.swt.BatchJobUI$1.widgetSelected(BatchJobUI.java:200)
                at org.eclipse.swt.widgets.TypedListener.handleEvent(TypedListener.java:249)
                at org.eclipse.swt.widgets.EventTable.sendEvent(EventTable.java:86)
                at org.eclipse.swt.widgets.Display.sendEvent(Display.java:4118)
                at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1052)
                at org.eclipse.swt.widgets.Display.runDeferredEvents(Display.java:3931)
                at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:3534)
                at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine$5.run(PartRenderingEngine.java:1170)
                at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:336)
                at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.run(PartRenderingEngine.java:1059)
                at org.eclipse.e4.ui.internal.workbench.E4Workbench.createAndRunUI(E4Workbench.java:153)
                at org.eclipse.ui.internal.Workbench.lambda$3(Workbench.java:667)
                at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:336)
                at org.eclipse.ui.internal.Workbench.createAndRunWorkbench(Workbench.java:597)
                at org.eclipse.ui.PlatformUI.createAndRunWorkbench(PlatformUI.java:148)
                at org.eclipse.chemclipse.rcp.app.ui.internal.support.ApplicationSupportDefault.start(ApplicationSupportDefault.java:26)
                at org.eclipse.chemclipse.rcp.app.ui.Application.start(Application.java:61)
                at org.eclipse.equinox.internal.app.EclipseAppHandle.run(EclipseAppHandle.java:196)
                at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.runApplication(EclipseAppLauncher.java:134)
                at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.start(EclipseAppLauncher.java:104)
                at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:388)
                at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:243)
                at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
                at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
                at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
                at java.lang.reflect.Method.invoke(Method.java:498)
                at org.eclipse.equinox.launcher.Main.invokeFramework(Main.java:656)
                at org.eclipse.equinox.launcher.Main.basicRun(Main.java:592)
                at org.eclipse.equinox.launcher.Main.run(Main.java:1498)
```

--- a/openchrom/plugins/net.openchrom.xxd.process.supplier.templates.ui/src/net/openchrom/xxd/process/supplier/templates/ui/wizards/PeakDetectorSupport.java
+++ b/openchrom/plugins/net.openchrom.xxd.process.supplier.templates.ui/src/net/openchrom/xxd/process/supplier/templates/ui/wizards/PeakDetectorSupport.java
@@ -1,5 +1,5 @@
 /*******************************************************************************
- * Copyright (c) 2019, 2020 Lablicate GmbH.
+ * Copyright (c) 2019, 2022 Lablicate GmbH.
  * 
  * All rights reserved. This program and the accompanying materials
  * are made available under the terms of the Eclipse Public License v1.0
@@ -50,13 +50,17 @@ public class PeakDetectorSupport {
 		wizardDialog.setMinimumPageSize(PeakDetectorWizard.DEFAULT_WIDTH, PeakDetectorWizard.DEFAULT_HEIGHT);
 		//
 		IProcessingInfo processingInfo = processSettings.getProcessingInfo();
-		int status = wizardDialog.open();
-		if(status == Dialog.OK) {
-			processingInfo.addInfoMessage(DESCRIPTION, "Successfully modified/added the peak(s).");
-		} else if(status == Dialog.CANCEL) {
-			processingInfo.addWarnMessage(DESCRIPTION, "Cancel has been pressed. No peak(s) added.");
-		} else {
-			processingInfo.addErrorMessage(DESCRIPTION, "Something went wrong to add the peak(s).");
+		try {
+			int status = wizardDialog.open();
+			if(status == Dialog.OK) {
+				processingInfo.addInfoMessage(DESCRIPTION, "Successfully modified/added the peak(s).");
+			} else if(status == Dialog.CANCEL) {
+				processingInfo.addWarnMessage(DESCRIPTION, "Cancel has been pressed. No peak(s) added.");
+			} else {
+				processingInfo.addErrorMessage(DESCRIPTION, "Something went wrong to add the peak(s).");
+			}
+		} finally {
+			wizard.dispose();
 		}
 	}
 }

--- a/openchrom/plugins/net.openchrom.xxd.process.supplier.templates.ui/src/net/openchrom/xxd/process/supplier/templates/ui/wizards/PeakReviewSupport.java
+++ b/openchrom/plugins/net.openchrom.xxd.process.supplier.templates.ui/src/net/openchrom/xxd/process/supplier/templates/ui/wizards/PeakReviewSupport.java
@@ -1,5 +1,5 @@
 /*******************************************************************************
- * Copyright (c) 2020 Lablicate GmbH.
+ * Copyright (c) 2020, 2022 Lablicate GmbH.
  * 
  * All rights reserved. This program and the accompanying materials
  * are made available under the terms of the Eclipse Public License v1.0
@@ -49,13 +49,17 @@ public class PeakReviewSupport {
 		wizardDialog.setMinimumPageSize(PeakReviewWizard.DEFAULT_WIDTH, PeakReviewWizard.DEFAULT_HEIGHT);
 		//
 		IProcessingInfo processingInfo = processSettings.getProcessingInfo();
-		int status = wizardDialog.open();
-		if(status == Dialog.OK) {
-			processingInfo.addInfoMessage(DESCRIPTION, "Successfully reviewed the peak(s).");
-		} else if(status == Dialog.CANCEL) {
-			processingInfo.addWarnMessage(DESCRIPTION, "Cancel has been pressed. No peak(s) reviewed.");
-		} else {
-			processingInfo.addErrorMessage(DESCRIPTION, "Something went wrong to review the peak(s).");
+		try {
+			int status = wizardDialog.open();
+			if(status == Dialog.OK) {
+				processingInfo.addInfoMessage(DESCRIPTION, "Successfully reviewed the peak(s).");
+			} else if(status == Dialog.CANCEL) {
+				processingInfo.addWarnMessage(DESCRIPTION, "Cancel has been pressed. No peak(s) reviewed.");
+			} else {
+				processingInfo.addErrorMessage(DESCRIPTION, "Something went wrong to review the peak(s).");
+			}
+		} finally {
+			wizard.dispose();
 		}
 	}
 }

